enumerables.rb

```diff
@@ -0,0 +1,66 @@
+# enumerable.rb
+module Enumerable
+  def my_each
+    i = 0
+    while to_a.length > i
+      yield(to_a[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while to_a.length > i
+      yield(to_a[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    new_array = []
+    to_a.my_each { |item| new_array.push(item) if yield(item) }
+    new_array
+  end
+
+  def my_all?
+    to_a.my_each { |item| return false if yield(item) == false }
+    true
+  end
+
```

Comment:
1. enumerables #my_all? when no block or argument is given returns true when none of the collection members are false or nil
2. enumerables #my_all? when no block or argument is given returns false when one of the collection members are false or nil
3. enumerables #my_all? when a class is passed as an argument returns true if all of the collection is a member of such class::Integer
4. enumerables #my_all? when a class is passed as an argument returns true if all of the collection is a member of such class::Numeric
5.  enumerables #my_all? when a class is passed as an argument returns false if any of the collection is not a member of such class
6.  enumerables #my_all? when a Regex is passed as an argument returns true if all of the collection matches the Regex
7. enumerables #my_all? when a Regex is passed as an argument returns false if any of the collection does not match the Regex
8. enumerables #my_all? when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern
9. enumerables #my_all? when a pattern other than Regex or a Class is given returns false if any of the collection does not match the pattern

Reply:
Solved all the requested changes and also tested it